agent/config/config.go

```diff
@@ -311,7 +311,20 @@ func environmentConfig() (Config, error) {
 	taskENIEnabled := utils.ParseBool(os.Getenv("ECS_ENABLE_TASK_ENI"), false)
 	taskIAMRoleEnabled := utils.ParseBool(os.Getenv("ECS_ENABLE_TASK_IAM_ROLE"), false)
 	taskIAMRoleEnabledForNetworkHost := utils.ParseBool(os.Getenv("ECS_ENABLE_TASK_IAM_ROLE_NETWORK_HOST"), false)
-	taskCPUMemLimitEnabled := utils.ParseBool(os.Getenv("ECS_ENABLE_TASK_CPU_MEM_LIMIT"), true)
+
+	var taskCPUMemLimitEnabled Conditional
+	taskCPUMemLimitConfigString := os.Getenv("ECS_ENABLE_TASK_CPU_MEM_LIMIT")
+
+	// We only want to set taskCPUMemLimit if it is explicitly set to true or false.
```

Comment:
I think it'd be a lot simpler to do this:

```go
if taskCPUMemLimitConfigString != "" {
	if utils.ParseBool(taskCPUMemLimitConfigString, false) {
		taskCPUMemLimitEnabled = ExplicitlyEnabled
	} else {
		taskCPUMemLimitEnabled = ExplicitlyDisabled
	}
}
```

Reply:
I have no idea why I didn't see it that way in the first place.